react-app/src/components/Extract.js

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { LinkContainer } from 'react-router-bootstrap'
+import { Col, Form, Row } from 'react-bootstrap'
+
+import '../stylesheets/Page.css';
+
+class Extract extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            configPath: "",
+            logPath: "",
+            includeDebug: false,
+            filterStart: false,
+            filterEnd: false
+        }
+
+        this.onConfigFileChange = this.onConfigFileChange.bind;
+        this.onLogFileChange = this.onLogFileChange.bind;
+        this.onIncludeDebugChange = this.onIncludeDebugChange.bind;
+        this.onFilterStartChange = this.onFilterStartChange.bind;
+        this.onFilterEndChange = this.onFilterEndChange.bind;
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    onConfigFileChange(e) {
```

Comment:
```suggestion
    onConfigFileChange = (e) => {
```

Using an arrow function instead makes the `this` keyword a valid thing inside the scope of this function

This should be repeated for all of the functions in this component.